### PR TITLE
Pet-37 add product entity and operations to backend

### PR DESCRIPTION
**Pull request by karimfathy054**

### CODE REVIEW

File: PVSSpringBoot/src/main/java/com/example/PVSSpringBoot/ControllerPackage/ProductController.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+<<<<<<< HEAD
+    @GetMapping("/all")
+=======
+    @GetMapping()
+>>>>>>> b6ee88146414874a7d00d51c8072ca0a820084f2
+    List<Product> findAll(){
+        return productManagementService.findAll();
+    }
+
+    @GetMapping("/hello")
```
Comment (OmarMahmoud11) [inline PVSSpringBoot/src/main/java/com/example/PVSSpringBoot/ControllerPackage/ProductController.java:29]: please remove this request

File: PVSSpringBoot/src/main/java/com/example/PVSSpringBoot/Entities/Product.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    @Column(name = "id", nullable = false)
+    private Long id;
+
+    @Column(name = "product_name", nullable = false)
+    @NonNull
+    private String productName;
+
+    @Column(name = "brand_name", nullable = false)
+    @NonNull
+    private String brandName;
```
Comment (OmarMahmoud11) [inline PVSSpringBoot/src/main/java/com/example/PVSSpringBoot/Entities/Product.java:28]: What do you mean by brand name?

File: PVSSpringBoot/src/main/java/com/example/PVSSpringBoot/ControllerPackage/ProductController.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    String hello(){
+        return "Hello";
+    }
+
+    @GetMapping("/id={id}")
+    Product findProduct(@PathVariable Long id){
+        return productManagementService.findById(id);
+    }
+
+    @GetMapping("/brand/{brandName}")
```
Comment (OmarMahmoud11) [inline PVSSpringBoot/src/main/java/com/example/PVSSpringBoot/ControllerPackage/ProductController.java:39]: Do you mean by this request filter ?
if no add a filter request.

**Review by OmarMahmoud11 — CHANGES_REQUESTED** [review]

File: PVSSpringBoot/src/main/java/com/example/PVSSpringBoot/Entities/Product.java (hunk 2 above)
Comment (karimfathy054) [reply PVSSpringBoot/src/main/java/com/example/PVSSpringBoot/Entities/Product.java:28]: a brand that can have multiple products with different names and uses (ex:Friskies producces both cat food and dog food) + was in the ERD

**Pull request merged**